### PR TITLE
Allow mutation of data buffer in RxToken

### PR DESCRIPTION
**Pull request by jhwgh1968**

### CODE REVIEW

File: src/phy/loopback.rs [hunk 1]
```diff
@@ -63,8 +63,12 @@ pub struct RxToken {
 }
 
 impl phy::RxToken for RxToken {
-    fn consume<R, F: FnOnce(&[u8]) -> Result<R>>(self, _timestamp: Instant, f: F) -> Result<R> {
-        f(&self.buffer)
+    fn consume<R, F: FnOnce(&mut [u8]) -> Result<R>>(
+        mut self,
+        _timestamp: Instant,
+        f: F
```
Comment (whitequark) [inline src/phy/loopback.rs]: This is a bit silly, but can you add a `where` clause here? I hate how you go one character beyond the line wrap and the Rust tools instantly explode everything to live on its own line.

**Comment by whitequark** [comment]
Looks like the doctests in phy/mod.rs fail?

**Comment by jhwgh1968** [comment]
Fixed, @whitequark. The CI is now green.

**Comment by whitequark** [comment]
@m-labs-homu r+

**Comment by m-labs-homu** [comment]
:pushpin: Commit 5a67d44 has been approved by `whitequark`

**Comment by whitequark** [comment]
Thanks!

**Comment by whitequark** [comment]
@m-labs-homu retry

**Comment by whitequark** [comment]
@m-labs-homu r+

**Comment by m-labs-homu** [comment]
:bulb: This pull request was already approved, no need to approve it again.

**Comment by m-labs-homu** [comment]
:pushpin: Commit 5a67d44 has been approved by `whitequark`

**Comment by m-labs-homu** [comment]
:hourglass: Testing commit 5a67d44e166b3eb5872bf54a0806ceb2c415a608 with merge f3cf4f61f499b4f86916071aeb9743d9f187e0e6...

**Comment by m-labs-homu** [comment]
:boom: Test timed out

**Comment by whitequark** [comment]
@m-labs-homu retry

**Comment by m-labs-homu** [comment]
:hourglass: Testing commit 5a67d44e166b3eb5872bf54a0806ceb2c415a608 with merge e6a6b1630fcc4af58f928b95bf88e005abb6c586...

**Comment by m-labs-homu** [comment]
:sunny: Test successful - [status-travis](https://travis-ci.org/m-labs/smoltcp/builds/528248009?utm_source=github_status&utm_medium=notification)
Approved by: whitequark
Pushing e6a6b1630fcc4af58f928b95bf88e005abb6c586 to master...

**Pull request closed without merging**